MeshGeoToolsLib/MeshNodeSearcher.cpp

```diff
@@ -154,10 +154,26 @@ MeshNodeSearcher& MeshNodeSearcher::getMeshNodeSearcher(
     if (_mesh_node_searchers.size() < mesh_id+1)
         _mesh_node_searchers.resize(mesh_id+1);
 
-    if (!_mesh_node_searchers[mesh_id])
+    if (_mesh_node_searchers[mesh_id])
+    {
+        // recreate searcher if search length algorithm does not fit
```

Comment:
Maybe my comment in the code is misleading. The existing searcher is only returned when it is from the same type (fixed or heuristic) and in the fixed case has the same internal search length. Else it is recreated. I will modify the comment ...